### ISSUE TITLE
SPIKE + Fix: Changes to the location search parameter in a Community providers search traps the user 

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-cms#15664]
## Describe the defect
When using the Facility Locator, the Service type field is required when "Community providers (in VA's network) is selected as the Facility type. If changes are made only to the geographic search parameter, the search cannot be completed until the Service type field is cleared and input is re-entered. 

## To Reproduce
1. Go to Facility Locator
2. Enter "West Virginia" as the geographic search parameter.
3. Select "Community providers (in VA's network) as Facility type
4. Select "Counselor - Professional" as Service type 
3. Click Search.
4. After search results are returned, edit the geographic search parameter to be "Morgantown, West Virginia"
5. Click Search
6. Observe the service field error prompting the user to search for an available service even though the field retained the previously selected service type. 

This behavior can also be triggered with these steps
1. Go to Facility Locator
2. Enter "West Virginia" as the geographic search parameter.
3. Select "Community providers (in VA's network) as Facility type
4. Select "Counselor - Professional" as Service type 
3. Click Search.
4. After search results are returned, zoom out and/or move the map to bring Morgantown into the center of the map
5. Observe "Search this area of the map" is grayed out and cannot be used
6. Click "Search" in the search box above the map
7. Observe the service field error prompting the user to search for an available service even though the field retained the previously selected service type. 
8. Zoom map in to place where a search should be permitted based on radius and observe the "Search this area of the map" is grayed out and still cannot be used 
9. Resolve service field error by clearing the field and re-selecting "Counselor - Professional"
10. Observe the "Search this area of the map" option is again available

## User experience flow

Depending on UX best practices and accessibility considerations, the service type field should probably either 

- honor the previously entered parameter which is still displayed (preferred) _or_
- clear the field and prompt the Veteran to search for an available service type

We may also need different error messaging to support a Veteran in resolving an error in the search field before attempting to use  the "Search this area of the map" option

We need to understand technically what is happening with the PPMS query in order to know whether either or both of these UX paths are possible. If both are technically possible, UX can determine preferred path forward.

## ACs

- [ ] Root cause analysis for what's happening to break search currently
- [ ] Review with UX to determine proper user flow within what's technically feasible
- [ ] User is able to change Facility Locator geo view to search a different geography for the same Community Care service successfully 

## Screenshots

<details>
<summary> Screenshot of error after geographic search parameters are changed </summary>

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOHN1QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ce55557a55d2e6dcd367dba8e111cc56623c359a/image.png)

</details>

<details>
<summary> Screenshot of error after interaction with the map </summary>

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBK0V1QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--7955b491fcd6ac8093132d848f84d5c5ad0e9b9a/image.png)
</details>

 ## Additional context
Add any other context about the problem here. Reach out to the Product Managers to determine if it should be escalated as critical (prevents users from accomplishing their work with no known workaround and needs to be addressed within 2 business days).

**Comment by mmiddaugh** [comment]
@laflannery when you have a moment, will you please evaluate this one for severity and make expected behavior recommendations?

**Comment by jilladams** [comment]
Would like to repro after component updates in FL to confirm it still occurs.

**Comment by jilladams** [comment]
Analytics-wise: 6% of searches are community care searches, so small proportion of traffic may hit this (Michelle to verify)

**Comment by davidmpickett** [comment]
I'd suggest that rather than trying to pull this into Sprint 2, we let Jordan complete her [redesign](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17835) of the Facility Locator search and have her account for this situation.

**Comment by mmiddaugh** [comment]
[Uploading... RPReplay_Final1713885944.mov]
I recorded this experience in mobile although it also occurs on desktop. 
User is able to resolve the issue via backspacing the text entry in the service type field but this workaround may not be immediately evident to all users. 
[Private Zenhub Video](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOXFHQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--3f243e0f3e837e4c2b952c7ace6dd0e48d6b7280/RPReplay_Final1713885944.mov)

**Comment by jilladams** [comment]
Noting; this task has become a partial SPIKE to understand the mechanism + conversation with UX to define fix + fix. 5 estimate is optimistic -- if the SPIKE portion takes the full 5, we will break implementation out to its own ticket.

**Comment by davidmpickett** [comment]
Pulling off UX label just for capacity clarity as the points here are mostly FE

**Comment by jilladams** [comment]
From planning: 
* We need to sort out root cause in order to determine best path to fix + effort for that fix. So: 5 pts is a rough guess, until we know more. 
* We didn't opt to break the fix into a separate ticket at this time, but may need to rethink that once we know root cause / UX fix.

**Comment by jilladams** [comment]
Max is going to help take look here.

**Comment by jilladams** [comment]
Assignment pivot: https://dsva.slack.com/archives/C0FQSS30V/p1715208560396599 🛋️

**Comment by maxx1128** [comment]
I've begun looking at this ticket. I've been able to reproduce the bug locally and believe the issue is that, while the service type is saved in the URL, the data isn't consistently saved in the local state. Plus, the `locationChanged` variable that effects if the error shows is set to `false` in this instance, even though it doesn't. Still looking into what the exact cause of these issues are.

**Comment by maxx1128** [comment]
Update for end-of-sprint: The code has been approved, but the tugboat instance has repeatedly failed. I have a screen recording of the updated code on the pull request and contacted Laura on Slack about helping her check the change locally. Most of this work is done and should be merged soon, all things being equal.

**Comment by davidmpickett** [comment]
PR has merged, needs prod verification

**Comment by laflannery** [comment]
Confirm on Prod that I can Search for a Community Provider then change my location and search again without the Service Type erroring - It now correctly remembers my original search term
